### PR TITLE
[develop2] conanfile_txt

### PR DESCRIPTION
**Pull request by memsharded**
Docs for https://github.com/conan-io/conan/pull/12699

### CODE REVIEW

File: reference/conanfile_txt.rst [hunk 1]
```diff
@@ -0,0 +1,141 @@
+.. _conanfile_txt_reference:
+
+conanfile.txt
+=============
+
+The ``conanfile.txt`` is a simplified version of ``conanfile.py``, aimed at simple consumption of dependencies, but it
```
Comment (RubenRBS) [inline reference/conanfile_txt.rst:6]:
```suggestion
The ``conanfile.txt`` file is a simplified version of ``conanfile.py``, aimed at simple consumption of dependencies, but it
```

File: reference/conanfile_txt.rst [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+cannot be used to create a package. Also, it is not necessary to have a ``conanfile.txt`` for consuming dependencies, 
+a ``conanfile.py`` can be perfectly use for simple consumption of dependencies.
```
Comment (RubenRBS) [inline reference/conanfile_txt.rst:8]:
```suggestion
a ``conanfile.py`` is perfectly suited for simple consumption of dependencies.
```

File: reference/conanfile_txt.rst [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+    7zip/16.00
+    cmake/3.23.0
+
+This section also supports version ranges and pinned recipe revisions, as above.
+
+In practice the ``[tool_requires]`` will be always installed (same as ``[requires]``) as installing from a *conanfile.txt* means that
+something is going to be built, so the tool requirements are indeed needed.
+Note however, that by default ``tool_requires`` live in the "build" context, they cannot be libraries to built with, just executable
+tools, and for example, using the ``CMakeDeps`` generator, they will not create CMake config files for them (an exception is possible,
+but it requires using a ``conanfile.py``, read the ``CMakeDeps`` reference for more information).
```
Comment (RubenRBS) [inline reference/conanfile_txt.rst:61]:
```suggestion
but it requires using a ``conanfile.py``, read the :ref:`CMakeDeps reference<CMakeDeps>` for more information).
```

File: reference/conanfile_txt.rst [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+a ``conanfile.py`` can be perfectly use for simple consumption of dependencies.
+
+It also provides a simplified functionality, for example it is not possible to express conditional requirements in
+``conanfile.txt``, and it will be necessary to use a ``conanfile.py`` for that. Read :ref:`consuming_packages_flexibility_of_conanfile_py`
+for more information about this.
+
+[requires]
+----------
+
+List of requirements, specifying the full reference. Equivalent to ``self.requires(<ref>)`` in the ``conanfile.py``.
```
Comment (RubenRBS) [inline reference/conanfile_txt.rst:17]:
```suggestion
List of requirements, specifying the full reference. Equivalent to ``self.requires(<ref>)`` in ``conanfile.py``. 
```

File: reference/conanfile_txt.rst [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+This section also supports version ranges and pinned recipe revisions, as above.
+The behavior of ``test_requires`` is totally equivalent to the ``[requires]`` section above, as the only difference is that
+``test_requires`` are not propagated to consumers, but as a ``conanfile.txt`` is never creating a package that can be consumed, it is
+irrelevant. It is provided to maintain the equivalence with ``conanfile.py``
+
+
+[generators]
+------------
+
+List of built-in generators to be used, equivalent to ``conanfile.py`` ``generators = "CMakeDeps", ...`` attribute.
```
Comment (RubenRBS) [inline reference/conanfile_txt.rst:84]:
```suggestion
List of built-in generators to be used, equivalent to the ``conanfile.py`` ``generators = "CMakeDeps", ...`` attribute.
```

**Review by RubenRBS — APPROVED** [review]
Looks great

**Review by czoido — APPROVED** [review]
Great!

**Pull request merged**